pkg/collector/upgrade/upgrade.go

```diff
@@ -108,7 +108,18 @@ func (u VersionUpgrade) ManagedInstance(ctx context.Context, otelcol v1alpha1.Op
 	}
 
 	if instanceV.GreaterThan(&Latest.Version) {
-		u.Log.Info("skipping upgrade for OpenTelemetry Collector instance, as it's newer than our latest version", "name", otelcol.Name, "namespace", otelcol.Namespace, "version", otelcol.Status.Version, "latest", Latest.Version.String())
+		// Update with the latest known version, which is what we have from versions.txt
+		u.Log.Info("no upgrade routines are needed for the OpenTelemetry instance", "name", otelcol.Name, "namespace", otelcol.Namespace, "version", otelcol.Status.Version, "latest", Latest.Version.String())
+
+		otelColV, err := semver.NewVersion(u.Version.OpenTelemetryCollector)
+		if err != nil {
+			return otelcol, err
+		}
+		if instanceV.LessThan(otelColV) {
+			u.Log.Info("upgraded OpenTelemetry Collector version", "name", otelcol.Name, "namespace", otelcol.Namespace, "version", otelcol.Status.Version)
+			otelcol.Status.Version = u.Version.OpenTelemetryCollector
+		}
```

Comment:
The 2 logs seem to contradict each other.

Reply:
Oh I missed the `"routines`" in the first log.  

In that case I think we should add a log for when we don't change anything.  We want to keep the existing "skipping upgrade for OpenTelemetry Collector instance" log for when we change nothing.

Reply:
We only change the version if `if instanceV.LessThan(otelColV)` is true right?  If it is false, then we follow the same logic as before where we could log `"skipping upgrade for OpenTelemetry Collector instance"`.